src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -89,16 +102,35 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
                 fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.LocalPath);
                 return fileDocumentIds;
             }
+            else if (documentUri.Scheme == SourceGeneratedFileScheme)
+            {
+                // In this case, the "host" portion is just the ProjectId directly.
+                var projectId = ProjectId.CreateFromSerialized(Guid.ParseExact(documentUri.Host, SourceGeneratedGuidFormat));
+
+                // The AbsolutePath will consist of a leading / to ignore, then the GUID that is the DocumentId, and then another slash, then the hint path
+                var slashAfterId = documentUri.AbsolutePath.IndexOf('/', startIndex: 1);
+                Contract.ThrowIfFalse(slashAfterId > 0, $"The URI '{documentUri}' is not formatted correctly.");
```

Comment:
is throwing correct here in LSP?  my guess is yes since this is arg validation of what the client is sending us..  checking with @dibarbet though.

Reply:
We've thrown in other places like this too. If that's not the pattern, then we have other places to update!

Reply:
if we have a source generator URI and we can't parse it, I think we should throw.